client/src/components/HolidayShow.js

```diff
@@ -78,37 +78,98 @@ const HolidayShow = (props) => {
         throw error
       }
       const respBody = await response.json()
-      const filteredReviews = holiday.reviews.filter((review) => review.id !== reviewId )
+      const filteredReviews = holiday.reviews.filter((review) => review.id !== reviewId)
       setErrors([])
       setHoliday({ ...holiday, reviews: filteredReviews })
     } catch (error) {
       console.log(`Error in fetch: ${error.message}`)
     }
   }
 
+  const editReview = async (reviewBody, reviewId) => {
+    try {
+      const response = await fetch(`/api/v1/reviews/${reviewId}`, {
+        method: "PATCH",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(reviewBody)
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json()
+          const updatedReviewsWithErrors = holiday.reviews.map((review) => {
+            if (review.id === reviewId) {
+              review.errors = body
+            }
+            return review
+          })
+          setHoliday({...holiday, reviews: updatedReviewsWithErrors})
```

Comment:
We are setting holidays here with errors, but what we would prefer to do is have a state to handle for edit errors and then set the errors to that state, which will also allow us to display errors to the user on the edit form